server/fleet/errors.go

```diff
@@ -304,6 +304,38 @@ func (e *MDMNotConfiguredError) Error() string {
 	return "MDM features aren't turned on in Fleet. For more information about setting up MDM, please visit https://fleetdm.com/docs/using-fleet/mobile-device-management"
 }
 
+// MDMBadGatewayError is an error type that generates a 502 status code.
+type MDMBadGatewayError struct {
```

Comment:
this looks great! is there a reason for the `MDM` prefix?

Reply:
I suppose not. Initially, I was thinking there might be some MDM-specific details or messaging we'd want to build in here, but it ended up being generic. I'll rename it without the MDM prefix.